### PR TITLE
Blog page accessibility fixes

### PR DESCRIPTION
**Pull request by youriwims**
Related PRs/issues #3934 #4004 

Primarily based fixes on Firefox devtools a11y label issue checker since a11y cli + ci is ongoing.

### CODE REVIEW

File: source/js/components/join/join.jsx [hunk 1]
```diff
@@ -327,14 +327,20 @@ export default class JoinUs extends React.Component {
...
           )}
-          <input
-            name="userEmail"
-            type="email"
-            className="form-control"
-            placeholder={getText(`Please enter your email`)}
-            ref={el => (this.email = el)}
-            onFocus={evt => this.onInputFocus(evt)}
-          />
+          <div
```
Comment (alanmoo) [inline source/js/components/join/join.jsx]: Why did you need to add a div here?

File: source/js/components/join/language-select.jsx [hunk 2]
```diff
@@ -36,13 +36,19 @@ export default class LanguageSelect extends React.Component {
     let classes = classNames(`form-control`, this.props.className);
 
     return (
-      <select
-        value={this.props.selectedLang}
-        onChange={evt => this.handleChange(evt)}
-        className={classes}
+      <div
+        id={`userLanguage-${this.props.formPosition}`}
```
Comment (alanmoo) [inline source/js/components/join/language-select.jsx]: What's this prop about and where does it get passed in?

File: source/js/components/join/join.jsx (hunk 1 above)
Comment (youriwims) [reply source/js/components/join/join.jsx]: Great question—so this `<input>` needs a label. I could've added `<label>`; however, our `<label>` have `display: inline-block` styling and I didn't want to have to do any style overrides to remove that without having a clear picture of what else that would effect. Simplest way to rememdy this without introducing any style changes was by using a `<div>` instead and adding `aria-label` attr. and `id`.

Screen Shot 2019-12-05 at 3 43 57 PM

File: source/js/components/join/language-select.jsx (hunk 2 above)
Comment (youriwims) [reply source/js/components/join/language-select.jsx]: It gets passed in [here](https://github.com/mozilla/foundation.mozilla.org/pull/4018/files#diff-bcd043d2cebc25df2ecfd6167302e3c8R398); I added that prop to the `id` since multiple `JoinUs` components can appear on a page and don't want any `id` errors 😬 .

Screen Shot 2019-12-05 at 3 55 55 PM

File: source/js/components/join/join.jsx (hunk 1 above)
Comment (alanmoo) [reply source/js/components/join/join.jsx]: But why couldn't you just add the "aria-label" directly to the `input` instead of the redirect through `aria-labeledby`?

Comment (youriwims) [reply source/js/components/join/join.jsx]: Because I've rarely seen that done 😛 But, I found a use-case for it [here](https://www.w3.org/TR/WCAG20-TECHS/ARIA14.html) so, I'll update the PR 👍

File: source/js/components/join/join.jsx [hunk 3]
```diff
@@ -334,6 +334,8 @@ export default class JoinUs extends React.Component {
             placeholder={getText(`Please enter your email`)}
             ref={el => (this.email = el)}
             onFocus={evt => this.onInputFocus(evt)}
+            aria-label={!this.isFlowForm() ? "Email" : ""}
+            id={this.isFlowForm() ? "userEmail-flow" : ""}
```
Comment (alanmoo) [inline source/js/components/join/join.jsx]: A little weird but I think I get it.
